File: apps/ios/Bedrud/Models/Room.swift
import Foundation

// MARK: - Room Settings

struct RoomSettings: Codable, Equatable, Sendable {
    let allowChat: Bool
    let allowVideo: Bool
    let allowAudio: Bool
    let requireApproval: Bool
    let e2ee: Bool

    static let `default` = RoomSettings(
        allowChat: true,
        allowVideo: true,
        allowAudio: true,
        requireApproval: false,
        e2ee: false
    )
}

// MARK: - Room Participant

struct RoomParticipant: Codable, Identifiable, Equatable, Sendable {
    let id: String
    let userId: String
    let email: String
    let name: String
    let joinedAt: String
    let isActive: Bool
    let isMuted: Bool
    let isVideoOff: Bool
    let isChatBlocked: Bool
    let permissions: String
}

// MARK: - Room

struct Room: Codable, Identifiable, Equatable, Sendable {
    let id: String
    let name: String
    let createdBy: String
    let adminId: String
    let isActive: Bool
    let isPublic: Bool
    let maxParticipants: Int
    let expiresAt: String
    let settings: RoomSettings
    let relationship: String?
    let mode: String
    let participants: [RoomParticipant]?

    static func == (lhs: Room, rhs: Room) -> Bool {
        lhs.id == rhs.id
    }
}

File: apps/android/app/src/main/java/com/bedrud/app/models/Room.kt
package com.bedrud.app.models

import com.google.gson.annotations.SerializedName

data class Room(
    val id: String,
    val name: String,
    @SerializedName("createdBy")
    val createdBy: String,
    @SerializedName("adminId")
    val adminId: String = "",
    @SerializedName("isActive")
    val isActive: Boolean = true,
    @SerializedName("isPublic")
    val isPublic: Boolean = true,
    @SerializedName("maxParticipants")
    val maxParticipants: Int = 50,
    @SerializedName("expiresAt")
    val expiresAt: String = "",
    val settings: RoomSettings = RoomSettings(),
    val relationship: String? = null,
    val mode: String = "meeting",
    val participants: List<RoomParticipant>? = null
)

data class RoomSettings(
    @SerializedName("allowChat")
    val allowChat: Boolean = true,
    @SerializedName("allowVideo")
    val allowVideo: Boolean = true,
    @SerializedName("allowAudio")
    val allowAudio: Boolean = true,
    @SerializedName("requireApproval")
    val requireApproval: Boolean = false,
    val e2ee: Boolean = false
)

data class RoomParticipant(
    val id: String,
    @SerializedName("userId")
    val userId: String,
    val email: String,
    val name: String,
    @SerializedName("joinedAt")
    val joinedAt: String,
    @SerializedName("isActive")
    val isActive: Boolean = true,
    @SerializedName("isMuted")
    val isMuted: Boolean = false,
    @SerializedName("isVideoOff")
    val isVideoOff: Boolean = false,
    @SerializedName("isChatBlocked")
    val isChatBlocked: Boolean = false,
    val permissions: String = ""
)
